core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-admin-page.component.spec.ts

```diff
@@ -0,0 +1,609 @@
+import { importAllAngularServices } from 'tests/unit-test-utils.ajs';
+// ^^^ This block is to be removed.
+
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { UserService } from 'services/user.service';
+import { UserInfo } from 'domain/user/user-info.model';
+import { ContributorDashboardAdminBackendApiService } from './services/contributor-dashboard-admin-backend-api.service';
+
+describe('Contributor dashboard admin page ', function() {
+  let $scope = null;
+  let ctrl = null;
+  let $rootScope = null;
+  let directive = null;
+  let userService: UserService = null;
+  let contributorDashboardAdminBackendApiService:
+    ContributorDashboardAdminBackendApiService = null;
+
+  let userInfo: UserInfo = null;
+
+  beforeEach(angular.mock.module('oppia'));
+  importAllAngularServices();
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+  });
+
+  beforeEach(angular.mock.inject(function($injector) {
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+
+    directive = $injector.get('contributorDashboardAdminPageDirective')[0];
+    userService = $injector.get('UserService');
+    contributorDashboardAdminBackendApiService = $injector.get(
+      'ContributorDashboardAdminBackendApiService');
+
+    userInfo = {
+      _roles: ['USER_ROLE'],
+      _isModerator: true,
+      _isCurriculumAdmin: false,
+      _isTopicManager: false,
+      _isSuperAdmin: false,
+      _canCreateCollections: true,
+      _preferredSiteLanguageCode: 'en',
+      _username: 'username1',
+      _email: 'tester@example.org',
+      _isLoggedIn: true,
+      isModerator: () => true,
+      isCurriculumAdmin: () => false,
+      isSuperAdmin: () => false,
+      isTopicManager: () => false,
+      isTranslationAdmin: () => true,
+      isQuestionAdmin: () => true,
+      canCreateCollections: () => true,
+      getPreferredSiteLanguageCode: () =>'en',
+      getUsername: () => 'username1',
+      getEmail: () => 'tester@example.org',
+      isLoggedIn: () => true
+    } as UserInfo;
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope,
+      $scope: $scope
+    });
+  }));
+
+  it('should fetch user info when initialized', fakeAsync(function() {
+    let userInfoSpy = spyOn(userService, 'getUserInfoAsync')
+      .and.returnValue(Promise.resolve(userInfo));
+
+    ctrl.$onInit();
+    tick();
+
+    expect(userInfoSpy).toHaveBeenCalled();
+  }));
+
+  describe('on clicking add rights button ', () => {
+    it('should successfully update the rights of the user', fakeAsync(() => {
+      let contributorDashboardAdminBackendApiServiceSpy = spyOn(
+        contributorDashboardAdminBackendApiService,
+        'addContributionReviewerAsync')
+        .and.returnValue(Promise.resolve(null));
+
```

Comment:
Can we remove new lines such that each test case has clear phases, e.g. setup, action, assertion? Here and below.

Reply:
updated here and elsewhere